inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/source/JdbcSource.java

```diff
@@ -0,0 +1,306 @@
+            }
+            CompletableFuture[] futureArray = futures.toArray(new CompletableFuture[futures.size()]);
+            CompletableFuture<Void> allFutures = CompletableFuture.allOf(futureArray);
+            allFutures.join();
+        }
+
+        /**
+         * Aggregate
+         * @param auditId
+         */
+        public void aggregate(String auditId) {
+            List<StartEndTime> statCycleList = sourceConfig.getAuditCycle() == DAY ? getStatCycleOfDay(statBackTimes)
+                    : getStatCycleOfMinute(statBackTimes, sourceConfig.getAuditCycle().getValue());
+            for (StartEndTime statCycle : statCycleList) {
+                long currentTimestamp = System.currentTimeMillis();
+                query(statCycle.getStartTime(), statCycle.getEndTime(), auditId);
+                long timeCost = System.currentTimeMillis() - currentTimestamp;
+                LOG.info("[{}]-[{}],{},stat back times:{},audit id:{},cost:{}ms",
+                        statCycle.getStartTime(), statCycle.getEndTime(),
+                        sourceConfig.getAuditCycle(),
+                        statBackTimes, auditId, timeCost);
+            }
+        }
+
+        /**
+         * Query
+         * @param startTime
+         * @param endTime
+         * @param auditId
+         */
+        public void query(String startTime, String endTime, String auditId) {
+            try {
+                Connection connection = dataSource.getConnection();
+                if (connection == null || connection.isClosed()) {
+                    createDataSource();
+                }
+                PreparedStatement pstat = connection.prepareStatement(sourceConfig.getQuerySql());
+                pstat.setString(1, startTime);
+                pstat.setString(2, endTime);
+                pstat.setString(3, auditId);
+                ResultSet resultSet = pstat.executeQuery();
+                while (resultSet.next()) {
+                    String inlongGroupID = resultSet.getString(1);
+                    String InlongStreamID = resultSet.getString(2);
+                    String AuditId = resultSet.getString(3);
+                    String AuditTag = resultSet.getString(4);
+                    long count = resultSet.getLong(5);
+                    long size = resultSet.getLong(6);
+                    long delay = resultSet.getLong(7);
+                    StatData data = new StatData();
+                    data.setLogTs(startTime);
+                    data.setInlongGroupId(inlongGroupID);
+                    data.setInlongStreamId(InlongStreamID);
+                    data.setAuditId(AuditId);
+                    data.setAuditTag(AuditTag);
+                    data.setCount(count);
+                    data.setSize(size);
+                    data.setDelay(delay);
+                    dataQueue.push(data);
+                }
+                resultSet.close();
+                pstat.close();
+                connection.close();
```

Comment:
Risk for resource leak, if an exception occurs before close.
Better to close in finally or use the try-with-resource to close resources automaticlly.